src/index.js

```diff
@@ -8,94 +8,39 @@ import makeE from './E';
 // import { HandledPromise, E } from '@agoric/eventual-send';
 // ...
 
-// TODO: Maybe rename the global HandledPromise to something only the tildot rewriter uses.
-if (!globalThis.HandledPromise) {
-  /* eslint-disable no-use-before-define */
-  // Install the shim as best we can.
-  maybeExtendPromise(Promise);
-  globalThis.HandledPromise = makeHandledPromise(Promise);
-  /* eslint-enable no-use-before-define */
+// eslint-disable-next-line import/no-mutable-exports
```

Comment:
Do you need `hp` to be a live binding? If not, I'd prefer

```js
const hp = typeof HandledPromise === 'undefined' ? makeHandledPromise() : HandledPromise;
```
or better
```js
const hp = typeof HandledPromise !== 'undefined' ?  HandledPromise : makeHandledPromise();
```

Then you could also get right of the disabling of this warning against live bindings.